src/main/java/net/minecraftforge/items/wrapper/PlayerArmorInvWrapper.java

```diff
@@ -0,0 +1,183 @@
+package net.minecraftforge.items.wrapper;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.player.InventoryPlayer;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.IChatComponent;
+
+public class PlayerArmorInvWrapper extends InvWrapper
+{
+    public final InventoryPlayer inventoryPlayer;
+    public final int offset;
+
+    public PlayerArmorInvWrapper(InventoryPlayer inv)
+    {
+        super(inv);
+
+        inventoryPlayer = inv;
+        offset = inventoryPlayer.mainInventory.length;
+    }
+
+    @Override
+    public int getSlots()
+    {
+        return inventoryPlayer.armorInventory.length;
+    }
+
+    @Override
+    public ItemStack getStackInSlot(int slot)
+    {
+        return super.getStackInSlot(slot + offset);
+    }
+
+    @Override
+    public ItemStack insertItem(int slot, ItemStack stack, boolean simulate)
+    {
+        // check if it's valid for the armor slot
+        if (slot < 4 && stack != null && stack.getItem().isValidArmor(stack, 3 - slot, inventoryPlayer.player))
+        {
+            return super.insertItem(slot + offset, stack, simulate);
+        }
+        return stack;
+    }
+
+    @Override
+    public void setStackInSlot(int slot, ItemStack stack)
+    {
+        super.setStackInSlot(slot + offset, stack);
+    }
+
+    @Override
+    public ItemStack extractItem(int slot, int amount, boolean simulate)
+    {
+        return super.extractItem(slot + offset, amount, simulate);
+    }
+
+    /**
+     * A simple wrapper that exposes the armor portion of the player inventory as its own IInventory and checks for validity
+     */
+    private static class ArmorInventoryWrapper implements IInventory
+    {
+
+        public final InventoryPlayer playerInventory;
```

Comment:
Seriously, why is this wrapping to a IInventory..... I don't like wrapping new stuff to old API....